### PR TITLE
Oauth lichess

### PR DESCRIPTION
**Pull request by kmandelbaum**
This PR
1. Adds the UI element to sign up/log in with Lichess.
2. Handles the lichess oauth flow with PKCE which is mandatory for lichess. This resulted in needing to store the `PkceCodeVerifier` (string) in the session data.
3. Adds `lichess_user_id` into the database, and corresponding harness.
4. Adds configuration options to support lichess oauth.

Caveats
- Sometimes when the user manages to click "Authorize" on the Lichess oauth page twice before they're redirected, they get an error because the token is already consumed. They will still be  logged in with their Lichess account.
- Lichess doesn't seem to be able to automatically redirect without user's explicit consent every time they want to log in. This is unlike Google that has the grant persistent in the user settings, and if the Google's own SSO cookie is present, that's enough for one-click login with Google. It is 2 clicks with Lichess every time ("Sign in with Lichess" on bughouse.pro and "Authorize" on lichess.org).

**Attention**
This requires a migration of the production `secret` database:

```
sqlite3 bughouse-secret.db "ALTER TABLE accounts ADD COLUMN lichess_user_id TEXT;"
```

### CODE REVIEW

File: src/session.rs [hunk 1]
```diff
@@ -38,20 +42,37 @@ pub struct GoogleOAuthRegistrationInfo {
...
+
+
 #[derive(Debug, Clone, Default, Serialize, Deserialize)]
 pub enum Session {
     Unknown,
     #[default]
     LoggedOut,
+    PkceChallengeInitiated(PkceChallengeInfo),
     GoogleOAuthRegistering(GoogleOAuthRegistrationInfo), // in the midst of Google OAuth signup
+    LichessOAuthRegistering(LichessOAuthRegistrationInfo), // in the midst of Google OAuth signup
```
Comment (amatveiakin) [inline src/session.rs:63]: typo: Google

File: www/index.html [hunk 2]
```diff
@@ -489,6 +492,15 @@
             <span class="flex-grow">Sign in with Google</span>
           </div>
         </button>
+        <button id="sign-with-lichess-button" class="big-button">
+          <div class="flex-row-full">
+		<svg id="lichess-logo" viewBox="0 0 80 80">
```
Comment (amatveiakin) [inline www/index.html:497]: tabs

File: www/index.js [hunk 3]
```diff
@@ -1717,6 +1724,9 @@ function update_session() {
     if (session.status == "google_oauth_registering") {
       push_menu_page(menu_signup_with_google_page);
     }
+    if (session.status == "lichess_oauth_registering") {
```
Comment (amatveiakin) [inline www/index.js:1727]: please also update the other place where "google_oauth_registering" is used

File: src/session.rs (hunk 1 above)
Comment (kmandelbaum) [reply src/session.rs:63]: Done

File: www/index.html (hunk 2 above)
Comment (kmandelbaum) [reply www/index.html:497]: Done

File: www/index.js (hunk 3 above)
Comment (kmandelbaum) [reply www/index.js:1727]: Done

File: www/index.js [hunk 4]
```diff
@@ -1654,6 +1654,12 @@ function update_session() {
         user_name = null;
         break;
       }
+      case "google_oauth_registering": {
```
Comment (amatveiakin) [inline www/index.js:1657]: This is google_oauth_registering again :)
Also I would collapse this into a single case, similarly to unknown and google_oauth_registering above.

Comment (kmandelbaum) [reply www/index.js:1657]: Whoops, done, thanks!

**Pull request merged**